### PR TITLE
feat(Actions): Automatically add labels on PRs

### PR DESCRIPTION
**Pull request by Kitzunu**
Automatically add Labels to PRs when they are opened depending on which files has been changed

### CODE REVIEW

**Comment by Kitzunu** [comment]
Well, this honestly just needs approval seeing as it works :)

**Comment by Kitzunu** [comment]
Love macOS...

File: .github/label-pr.yml
Comment (FrancescoBorzi) [inline .github/label-pr.yml:4]: This is wrong and reminds me that we should write this somewhere on our wiki.

By "Core" we mean everything that is:

- C++ **except** the contents of `/src/server/scripts`, which are instead "Scripts" (for historical reasons).
- DB structure changes. So SQL changes can be "Core" too.

File: .github/workflows/labeler.yml
Comment (FrancescoBorzi) [inline .github/workflows/labeler.yml:7]: why not using latest Ubuntu? `ubuntu-20.04`

Comment (Kitzunu) [reply .github/workflows/labeler.yml:7]: Because those who made the script use that one, dno

File: .github/label-pr.yml
Comment (Kitzunu) [reply .github/label-pr.yml:4]: well, if you do DB changes that count as core too, then it will probably have a core change as well.
This was more thought of helping people to more easily understand what type of PR it is and what way it would need to be tested

File: .github/workflows/labeler.yml
Comment (FrancescoBorzi) [reply .github/workflows/labeler.yml:7]: shall we try latest?

File: .github/label-pr.yml
Comment (FrancescoBorzi) [reply .github/label-pr.yml:4]: sure @Kitzunu , but can we exclude `/src/server/scripts` from applying "CORE" ?

Comment (Kitzunu) [reply .github/label-pr.yml:4]: Dont know

File: .github/workflows/labeler.yml
Comment (Kitzunu) [reply .github/workflows/labeler.yml:7]: why not ^^

File: .github/label-pr.yml
Comment (Kitzunu) [reply .github/label-pr.yml:4]: I have no idea if this will work. regEx is a regular expression

File: .github/label-pr.yml
Comment (Kitzunu) [inline .github/label-pr.yml:3]: @FrancescoBorzi

Comment (FrancescoBorzi) [reply .github/label-pr.yml:3]: @Kitzunu is this supposed to exclude the scripts? where is the part of "including all cpp/h files" ?

Comment (Kitzunu) [reply .github/label-pr.yml:3]: I just thought so too. Will look mord into it when Im not trying to fix it on my phone xd

**Review by FrancescoBorzi — APPROVED** [review]
@Kitzunu fine for me as long as you'll take care that this works as intended. It looks like there is no way to make sure this works until we merge it, so you can merge it!

**Comment by Kitzunu** [comment]
I think there is a way, like I did woth CORE now.

Gonna test it and draw a conclution after

**Comment by Kitzunu** [comment]
I cant figure out hwo to exclude it...

I have tried every guide there is....

**Comment by Kitzunu** [comment]
@FrancescoBorzi It isn't possible to exclude paths, but what if we add a new label called Scripts that only apply to them, so it will have CORE & Script?

**Comment by Kitzunu** [comment]
Now it works decently :)

**Pull request merged**
